matchbook-sdk-rest/src/main/java/com/matchbook/sdk/rest/configs/wrappers/HttpClientWrapper.java

```diff
@@ -34,14 +42,56 @@ public HttpClientWrapper(HttpConfig httpConfig) {
     }
 
     private OkHttpClient initHttpClient(HttpConfig httpConfig) {
+        CookieJar cookieJar = createCookieJar();
         return new OkHttpClient.Builder()
                 .connectTimeout(httpConfig.getConnectionTimeout(), TimeUnit.MILLISECONDS)
                 .writeTimeout(httpConfig.getWriteTimeout(), TimeUnit.MILLISECONDS)
                 .readTimeout(httpConfig.getReadTimeout(), TimeUnit.MILLISECONDS)
                 .followRedirects(false)
+                .cookieJar(cookieJar)
                 .build();
     }
 
+    /**
+     * Create a {@link CookieJar} to hold in memory the cookies. These are stored based on the host of the request.
+     *
+     * @return a {@link CookieJar} instance
+     */
+    private CookieJar createCookieJar() {
+        return new CookieJar() {
+            private final ConcurrentHashMap<String, List<Cookie>> cookieStore = new ConcurrentHashMap<>();
+
+            @Override
+            public void saveFromResponse(HttpUrl url, List<Cookie> cookies) {
+                if (cookies != null) {
+                    cookieStore.put(url.host(), cookies);
+                }
+            }
+
+            @Override
+            public List<Cookie> loadForRequest(HttpUrl url) {
+                List<Cookie> cookies = cookieStore.get(url.host());
+
+                if (cookies == null) {
+                    return new ArrayList<Cookie>();
```

Comment:
you don't need to specify `new ArrayList<Cookie>()`;, `just new ArrayList<>();` would be better

Reply:
Change pushed